lib/phoenix/controller.ex

```diff
@@ -627,8 +629,17 @@ defmodule Phoenix.Controller do
 
     view = Map.get(conn.private, :phoenix_view) ||
             raise "a view module was not specified, set one with put_view/2"
-    data = Phoenix.View.render_to_iodata(view, template,
-                                         Map.put(conn.assigns, :conn, conn))
+
+    data =
+      if endpoint = conn.private[:phoenix_endpoint] do
+        instrument endpoint, :phoenix_controller_render, template <> format, fn ->
```

Comment:
I would make `instrument`, `%Plug.Conn{}` aware. That way we can match on the conn, and if it has a phoenix_endpoint we do our thing, otherwise we just execute the block. 

```elixir
    data =
      instrument conn, :phoenix_controller_render, template <> format, fn ->
        Phoenix.View.render_to_iodata(view, template,
                                      Map.put(conn.assigns, :conn, conn))
      end
```